packages/web-components/fast-foundation/src/data-grid/data-grid.ts

```diff
@@ -184,6 +189,61 @@ export class FASTDataGrid extends FASTElement {
         }
     }
 
+    /**
+     * Defines how the grid handles row or cell selection.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: selection-mode
+     */
+    @attr({ attribute: "selection-mode" })
+    public selectionMode: DataGridSelectionMode = "none";
+    private selectionModeChanged(
+        prev: DataGridSelectionMode,
+        next: DataGridSelectionMode
+    ): void {
+        if (this.$fastController.isConnected) {
+            if (prev === "single-row" || prev === "multi-row") {
+                this.removeEventListener(
+                    "rowselectionchange",
+                    this.handleRowSelectedChange
+                );
+            }
+            if (next === "single-row" || next === "multi-row") {
+                this.addEventListener("rowselectionchange", this.handleRowSelectedChange);
+            }
+            this.deselectAllRows();
+        }
+    }
+
+    /**
+     * Determines if clicks can automatically select
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: click-select
+     */
+    @attr({ attribute: "click-select", mode: "boolean" })
+    public clickSelect: boolean = true;
```

Comment:
I actually think we'll need to rethink how to set this up with an attribute name so that the boolean value does not default to true (needs to default to false). See #5320 as reference.

Reply:
done, now "disable-click-select"